src/components/list/list-renderer.tsx

```diff
@@ -127,6 +127,7 @@ export class ListRenderer {
             'mdc-deprecated-list-item': true,
             'mdc-deprecated-list-item--disabled': item.disabled,
             'mdc-deprecated-list-item--selected': item.selected,
+            'has-primary-component': !!item?.primaryComponent,
```

Comment:
I suggest adding a function for checking if there is a primaryComponent, so that we're sure it's done the same way everywhere. I also suggest checking for `item?.primaryComponent?.name`.  I'll add a fixup commit with this suggestion though.